File: Python/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i_index, i in enumerate(nums):
            remain_nums = nums[i_index + 1:]
            t = target - i
            if t in remain_nums:
                return [i_index, remain_nums.index(t) + i_index + 1]

File: C++/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len_nums = nums.size();

        unordered_map<int, int> mp;
        for (int i=0; i<len_nums; i++){
            mp[nums[i]] = i;        // [數字, index]
        }
        
        for (int i=0; i<len_nums; i++){
            int t = target-nums[i];
            if (mp.find(t) != mp.end() && mp[t] != i){
                return {i, mp[t]};
            }
        }
        return {};
    }
};